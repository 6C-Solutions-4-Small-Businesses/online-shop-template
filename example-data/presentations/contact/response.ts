export const response = {
    'name': 'Demo BUSINESS',
    'email': 'contact@6csolutions.ca',
    'phoneNumber': '14501115555',
    'address': '1 Rue 7055 Taschereau Blvd, Brossard, QC J4Z 1A7, Canada',
    'openingHours': [{
        'dayRange': ['Monday', 'Wednesday'],
        'open': '10:00',
        'close': '18:00'
    }, {'dayRange': ['Thursday'], 'open': '10:00', 'close': '20:00'}, {
        'dayRange': ['Friday', 'Saturday'],
        'open': '10:00',
        'close': '21:00'
    }, {'dayRange': ['Sunday']}]
}
