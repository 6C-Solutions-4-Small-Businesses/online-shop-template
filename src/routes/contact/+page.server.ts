import type {BusinessContactPresentation} from '$lib/frontend/presentations/BusinessContactPresentation'
import {throwInternalServerError, throwInvalidRequestError} from '$lib/shared/ErrorsHandling'

export const prerender = true

export const load = async (event): Promise<BusinessContactPresentation> => {
    let response: Response
    try {
        response = await event.fetch('/api/v1/contact', {method: 'GET'})
    } catch (e) {
        throwInternalServerError({message: 'COULD_NOT_FETCH_BUSINESS_CONTACT', cause: e as Error})
    }

    if (!response.ok) {
        throwInvalidRequestError({message: 'COULD_NOT_FETCH_BUSINESS_CONTACT'})
    }

    return await response.json()
}



