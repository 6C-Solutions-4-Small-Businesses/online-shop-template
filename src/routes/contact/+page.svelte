<script lang="ts">
    import {onMount} from 'svelte'
    import type {PageData} from './$types'
    import * as GMaps from '@googlemaps/js-api-loader'
    import {PUBLIC_GOOGLE_MAPS_JAVASCRIPT_API_KEY} from '$env/static/public'
    import Button from '$lib/frontend/components/Button.svelte'
    import {contactBusiness} from '$lib/frontend/endpoints/Endpoints'
    import {getToastStore, type ToastSettings} from '@skeletonlabs/skeleton'
    import {getErrorToastSettings, getSuccessToastSettings} from '$lib/frontend/core/ToasterUtils'

    export let data: PageData

    const {Loader} = GMaps
    let googleMapElement: any
    let name: string
    let email: string
    let subject: string
    let message: string

    const toastStore = getToastStore()

    $: isSendButtonDisabled = !name || !email || !subject || !message

    async function initMap(): Promise<void> {
        const loader = new Loader({
            apiKey: PUBLIC_GOOGLE_MAPS_JAVASCRIPT_API_KEY,
            version: 'weekly',
        })

        let map: any

        loader.importLibrary('geocoding').then(({Geocoder}) => {
            const geocoder = new Geocoder()
            geocoder.geocode({address: data.address}, (results: any, status: any) => {
                if (status === 'OK') {
                    const location = results[0].geometry.location
                    loader.importLibrary('maps').then(({Map}) => {
                        map = new Map(googleMapElement, {
                            zoom: 15,
                            center: location,
                            fullscreenControl: false,
                            zoomControl: true,
                            streetViewControl: false,
                            mapId: 'business-location-map',
                        })
                    })
                    loader.importLibrary('marker').then(({AdvancedMarkerElement}) => {
                        new AdvancedMarkerElement({
                            position: location,
                            map,
                            title: data.name,
                        })
                    })
                } else {
                    toastStore.trigger(getErrorToastSettings('Une erreur est survenue durant l\'affichage de la carte.'))
                    console.log('Geocode was not successful for the following reason: ' + status)
                }
            })
        })
    }

    onMount(async () => {
        await initMap()
    })

    function clearContactUsForm(): void {
        name = ''
        email = ''
        subject = ''
        message = ''
    }

    async function contactBusinessHandler() {
        let toastSettings: ToastSettings

        const presentation = await contactBusiness({name, email, subject, message})
        if (presentation) {
            clearContactUsForm()
            toastSettings = getSuccessToastSettings('Votre message a bien été envoyé. Nous vous répondrons dans les plus brefs délais. Merci !')
        } else {
            toastSettings = getErrorToastSettings('Malheureusement, nous n\'avons pas pu envoyer votre message. Veuillez réessayer plus tard.')
        }

        toastStore.trigger(toastSettings)
    }
</script>

<div class="w-full max-w-full h-fit flex flex-col justify-center items-center bg-slate-100">
    <div class="w-full lg:w-9/12 bg-white py-6 mb-0 lg:my-5">
        <p class="title text-2xl text-center font-light">Contactez-nous</p>
        <div class="w-full h-full flex flex-col md:flex-row">
            <div class="w-full md:w-1/2 h-full flex flex-col justify-center items-center py-6 sm:py-14 gap-6">
                <div class="w-10/12 md:w-9/12 h-full flex flex-col gap-4">
                    <h4 class="text-xl uppercase">Où nous trouver</h4>
                    <div bind:this={googleMapElement} class="h-96 bg-slate-100"></div>
                </div>
                <div class="w-10/12 md:w-9/12 flex flex-col gap-4">
                    <h4 class="text-xl uppercase">Coordonnées</h4>
                    <div class="flex flex-col gap-2">
                        <div class="w-full md:w-7/12 text-md font-light">{data.address}</div>
                        <div class="w-full md:w-7/12 text-md font-light">{data.phoneNumber}</div>
                        <div class="w-full md:w-7/12 text-md font-light">{data.email}</div>
                    </div>
                </div>
                <div class="w-10/12 md:w-9/12 flex flex-col gap-4">
                    <h4 class="text-xl uppercase">Nous sommes ouverts</h4>
                    {#each data.openingHours as {dayRange, open, close}}
                        {#if (open && close)}
                            <div class="w-full lg:w-7/12 flex flex-col gap-2">
                                <div class="w-full flex font-light text-justify">
                                    <div class="w-7/12">{dayRange.join(' - ')}</div>
                                    <div class="w-5/12 text-end">{`${open} - ${close}`}</div>
                                </div>
                            </div>
                        {/if}
                    {/each}
                </div>
            </div>
            <div class="w-full md:w-1/2 flex flex-col py-6 sm:py-14 items-center gap-3">
                <div class="w-10/12 md:w-9/12 flex flex-col gap-2">
                    <h4 class="text-xl uppercase">Prenez contact avec nous !</h4>
                    <p class="text-justify font-light">
                        Si vous avez des questions ou si vous souhaitez prendre rendez-vous, veuillez utiliser le
                        formulaire de contact ci-dessous ou appelez-nous directement au
                        <a class="text-blue-600" href="tel:{data.phoneNumber}">{data.phoneNumber}</a>.
                    </p>
                </div>
                <div class="w-10/12 md:w-9/12 flex flex-col gap-3 font-light">
                    <div class="flex flex-col gap-2">
                        <div>Name<span class="text-red-500">*</span></div>
                        <input bind:value={name} class="h-10 border-2" required type="text"/>
                    </div>
                    <div class="flex flex-col gap-2">
                        <div>Email<span class="text-red-500">*</span></div>
                        <input bind:value={email} class="h-10 border-2" required type="email">
                    </div>
                    <div class="flex flex-col gap-2">
                        <div>Sujet<span class="text-red-500">*</span></div>
                        <input bind:value={subject} class="h-10 border-2" type="text">
                    </div>
                    <div class="flex flex-col gap-2">
                        <div>Message<span class="text-red-500">*</span></div>
                        <textarea bind:value={message} class="border-2" required rows="4"></textarea>
                    </div>
                    <div class="flex justify-end">
                        <Button
                                background="bg-orange-50 hover:bg-white"
                                border="border border-orange-500"
                                classNames="w-36 h-12 border border-white shadow-sm mb-2"
                                disabled={isSendButtonDisabled}
                                id="contact-business-button"
                                onClick={contactBusinessHandler}
                                text="text-orange-500"
                        >
                            Envoyer
                        </Button>
                    </div>
                </div>
            </div>
        </div>
    </div>
</div>

<style lang="css">
    .title {
        font-family: Roboto, sans-serif
    }
</style>
