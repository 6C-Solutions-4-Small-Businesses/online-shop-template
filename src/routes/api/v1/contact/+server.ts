import {json, type RequestHandler} from '@sveltejs/kit'
import {response} from "$data/presentations/contact/response";
import {FETCH_REQUESTS_DELAY} from "$routes/api/Config";

export const GET: RequestHandler = async ({request}) => {
    await new Promise((resolve) => setTimeout(resolve, FETCH_REQUESTS_DELAY))

    return json(response)
}
