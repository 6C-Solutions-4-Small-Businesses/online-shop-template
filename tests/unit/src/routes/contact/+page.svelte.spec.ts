import {render} from '@testing-library/svelte'
import {expect} from 'vitest'
import ContactPage from '$routes/contact/+page.svelte'
import {mock} from 'vitest-mock-extended'
import type {BusinessContactPresentation} from '$lib/frontend/presentations/BusinessContactPresentation'
import {getDemoMerchantPresentedOpeningHours} from '$tests/unit/src/routes/contact/ContactPageTestHelper'
import type { CategorySummaryPresentation } from '$lib/frontend/presentations/CategorySummaryPresentation'

describe('Contact Page', () => {

    describe('HTML', () => {

        it('should match snapshot', () => {
            const {container} = render(ContactPage, {
                data: mock<BusinessContactPresentation & {
                    categories: CategorySummaryPresentation[],
                }>({
                    openingHours: getDemoMerchantPresentedOpeningHours(),
                })
            })

            expect(container).toMatchSnapshot()
        })
    })
})
