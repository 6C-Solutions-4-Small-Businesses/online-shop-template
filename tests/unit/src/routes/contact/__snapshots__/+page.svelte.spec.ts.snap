// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Contact Page > HTML > should match snapshot 1`] = `
<body>
  <div>
    <div
      class="w-full max-w-full h-fit flex flex-col justify-center items-center bg-slate-100 s-2CpGgZ1sTTPV"
    >
      <div
        class="w-full lg:w-9/12 bg-white py-6 mb-0 lg:my-5 s-2CpGgZ1sTTPV"
      >
        <p
          class="title text-2xl text-center font-light s-2CpGgZ1sTTPV"
        >
          Contactez-nous
        </p>
         
        <div
          class="w-full h-full flex flex-col md:flex-row s-2CpGgZ1sTTPV"
        >
          <div
            class="w-full md:w-1/2 h-full flex flex-col justify-center items-center py-6 sm:py-14 gap-6 s-2CpGgZ1sTTPV"
          >
            <div
              class="w-10/12 md:w-9/12 h-full flex flex-col gap-4 s-2CpGgZ1sTTPV"
            >
              <h4
                class="text-xl uppercase s-2CpGgZ1sTTPV"
              >
                Où nous trouver
              </h4>
               
              <div
                class="h-96 bg-slate-100 s-2CpGgZ1sTTPV"
              />
            </div>
             
            <div
              class="w-10/12 md:w-9/12 flex flex-col gap-4 s-2CpGgZ1sTTPV"
            >
              <h4
                class="text-xl uppercase s-2CpGgZ1sTTPV"
              >
                Coordonnées
              </h4>
               
              <div
                class="flex flex-col gap-2 s-2CpGgZ1sTTPV"
              >
                <div
                  class="w-full md:w-7/12 text-md font-light s-2CpGgZ1sTTPV"
                >
                  function(...s) {
    let r = A(t);
    r.called = !0, r.callCount++, r.calls.push(s);
    let m = r.next.shift();
    if (m) {
      r.results.push(m);
      let [l, o] = m;
      if (l === "ok")
        return o;
      throw o;
    }
    let p, d = "ok";
    if (r.impl)
      try {
        new.target ? p = Reflect.construct(r.impl, s, new.target) : p = r.impl.apply(this, s), d = "ok";
      } catch (l) {
        throw p = l, d = "error", r.results.push([d, l]), l;
      }
    let a = [d, p];
    if (b(p)) {
      let l = p.then((o) =&gt; a[1] = o).catch((o) =&gt; {
        throw a[0] = "error", a[1] = o, o;
      });
      Object.assign(l, p), p = l;
    }
    return r.results.push(a), p;
  }
                </div>
                 
                <div
                  class="w-full md:w-7/12 text-md font-light s-2CpGgZ1sTTPV"
                >
                  function(...s) {
    let r = A(t);
    r.called = !0, r.callCount++, r.calls.push(s);
    let m = r.next.shift();
    if (m) {
      r.results.push(m);
      let [l, o] = m;
      if (l === "ok")
        return o;
      throw o;
    }
    let p, d = "ok";
    if (r.impl)
      try {
        new.target ? p = Reflect.construct(r.impl, s, new.target) : p = r.impl.apply(this, s), d = "ok";
      } catch (l) {
        throw p = l, d = "error", r.results.push([d, l]), l;
      }
    let a = [d, p];
    if (b(p)) {
      let l = p.then((o) =&gt; a[1] = o).catch((o) =&gt; {
        throw a[0] = "error", a[1] = o, o;
      });
      Object.assign(l, p), p = l;
    }
    return r.results.push(a), p;
  }
                </div>
                 
                <div
                  class="w-full md:w-7/12 text-md font-light s-2CpGgZ1sTTPV"
                >
                  function(...s) {
    let r = A(t);
    r.called = !0, r.callCount++, r.calls.push(s);
    let m = r.next.shift();
    if (m) {
      r.results.push(m);
      let [l, o] = m;
      if (l === "ok")
        return o;
      throw o;
    }
    let p, d = "ok";
    if (r.impl)
      try {
        new.target ? p = Reflect.construct(r.impl, s, new.target) : p = r.impl.apply(this, s), d = "ok";
      } catch (l) {
        throw p = l, d = "error", r.results.push([d, l]), l;
      }
    let a = [d, p];
    if (b(p)) {
      let l = p.then((o) =&gt; a[1] = o).catch((o) =&gt; {
        throw a[0] = "error", a[1] = o, o;
      });
      Object.assign(l, p), p = l;
    }
    return r.results.push(a), p;
  }
                </div>
              </div>
            </div>
             
            <div
              class="w-10/12 md:w-9/12 flex flex-col gap-4 s-2CpGgZ1sTTPV"
            >
              <h4
                class="text-xl uppercase s-2CpGgZ1sTTPV"
              >
                Nous sommes ouverts
              </h4>
               
              <div
                class="w-full lg:w-7/12 flex flex-col gap-2 s-2CpGgZ1sTTPV"
              >
                <div
                  class="w-full flex font-light text-justify s-2CpGgZ1sTTPV"
                >
                  <div
                    class="w-7/12 s-2CpGgZ1sTTPV"
                  >
                    Lundi
                  </div>
                   
                  <div
                    class="w-5/12 text-end s-2CpGgZ1sTTPV"
                  >
                    11:00 - 17:00
                  </div>
                </div>
                 
              </div>
              
              <div
                class="w-full lg:w-7/12 flex flex-col gap-2 s-2CpGgZ1sTTPV"
              >
                <div
                  class="w-full flex font-light text-justify s-2CpGgZ1sTTPV"
                >
                  <div
                    class="w-7/12 s-2CpGgZ1sTTPV"
                  >
                    Mardi - Vendredi
                  </div>
                   
                  <div
                    class="w-5/12 text-end s-2CpGgZ1sTTPV"
                  >
                    10:00 - 19:00
                  </div>
                </div>
                 
              </div>
              
              <div
                class="w-full lg:w-7/12 flex flex-col gap-2 s-2CpGgZ1sTTPV"
              >
                <div
                  class="w-full flex font-light text-justify s-2CpGgZ1sTTPV"
                >
                  <div
                    class="w-7/12 s-2CpGgZ1sTTPV"
                  >
                    Samedi
                  </div>
                   
                  <div
                    class="w-5/12 text-end s-2CpGgZ1sTTPV"
                  >
                    10:00 - 18:00
                  </div>
                </div>
                 
              </div>
              
              <div
                class="w-full lg:w-7/12 flex flex-col gap-2 s-2CpGgZ1sTTPV"
              >
                <div
                  class="w-full flex font-light text-justify s-2CpGgZ1sTTPV"
                >
                  <div
                    class="w-7/12 s-2CpGgZ1sTTPV"
                  >
                    Dimanche
                  </div>
                   
                  <div
                    class="w-5/12 text-end s-2CpGgZ1sTTPV"
                  >
                    11:00 - 17:00
                  </div>
                </div>
                 
              </div>
              
            </div>
          </div>
           
          <div
            class="w-full md:w-1/2 flex flex-col py-6 sm:py-14 items-center gap-3 s-2CpGgZ1sTTPV"
          >
            <div
              class="w-10/12 md:w-9/12 flex flex-col gap-2 s-2CpGgZ1sTTPV"
            >
              <h4
                class="text-xl uppercase s-2CpGgZ1sTTPV"
              >
                Prenez contact avec nous !
              </h4>
               
              <p
                class="text-justify font-light s-2CpGgZ1sTTPV"
              >
                Si vous avez des questions ou si vous souhaitez prendre rendez-vous, veuillez utiliser le
                        formulaire de contact ci-dessous ou appelez-nous directement au
                        
                <a
                  class="text-blue-600 s-2CpGgZ1sTTPV"
                  href="tel:function(...s) {
    let r = A(t);
    r.called = !0, r.callCount++, r.calls.push(s);
    let m = r.next.shift();
    if (m) {
      r.results.push(m);
      let [l, o] = m;
      if (l === "ok")
        return o;
      throw o;
    }
    let p, d = "ok";
    if (r.impl)
      try {
        new.target ? p = Reflect.construct(r.impl, s, new.target) : p = r.impl.apply(this, s), d = "ok";
      } catch (l) {
        throw p = l, d = "error", r.results.push([d, l]), l;
      }
    let a = [d, p];
    if (b(p)) {
      let l = p.then((o) => a[1] = o).catch((o) => {
        throw a[0] = "error", a[1] = o, o;
      });
      Object.assign(l, p), p = l;
    }
    return r.results.push(a), p;
  }"
                >
                  function(...s) {
    let r = A(t);
    r.called = !0, r.callCount++, r.calls.push(s);
    let m = r.next.shift();
    if (m) {
      r.results.push(m);
      let [l, o] = m;
      if (l === "ok")
        return o;
      throw o;
    }
    let p, d = "ok";
    if (r.impl)
      try {
        new.target ? p = Reflect.construct(r.impl, s, new.target) : p = r.impl.apply(this, s), d = "ok";
      } catch (l) {
        throw p = l, d = "error", r.results.push([d, l]), l;
      }
    let a = [d, p];
    if (b(p)) {
      let l = p.then((o) =&gt; a[1] = o).catch((o) =&gt; {
        throw a[0] = "error", a[1] = o, o;
      });
      Object.assign(l, p), p = l;
    }
    return r.results.push(a), p;
  }
                </a>
                .
              </p>
            </div>
             
            <div
              class="w-10/12 md:w-9/12 flex flex-col gap-3 font-light s-2CpGgZ1sTTPV"
            >
              <div
                class="flex flex-col gap-2 s-2CpGgZ1sTTPV"
              >
                <div
                  class="s-2CpGgZ1sTTPV"
                >
                  Name
                  <span
                    class="text-red-500 s-2CpGgZ1sTTPV"
                  >
                    *
                  </span>
                </div>
                 
                <input
                  class="h-10 border-2 s-2CpGgZ1sTTPV"
                  required=""
                  type="text"
                />
              </div>
               
              <div
                class="flex flex-col gap-2 s-2CpGgZ1sTTPV"
              >
                <div
                  class="s-2CpGgZ1sTTPV"
                >
                  Email
                  <span
                    class="text-red-500 s-2CpGgZ1sTTPV"
                  >
                    *
                  </span>
                </div>
                 
                <input
                  class="h-10 border-2 s-2CpGgZ1sTTPV"
                  required=""
                  type="email"
                />
              </div>
               
              <div
                class="flex flex-col gap-2 s-2CpGgZ1sTTPV"
              >
                <div
                  class="s-2CpGgZ1sTTPV"
                >
                  Sujet
                  <span
                    class="text-red-500 s-2CpGgZ1sTTPV"
                  >
                    *
                  </span>
                </div>
                 
                <input
                  class="h-10 border-2 s-2CpGgZ1sTTPV"
                  type="text"
                />
              </div>
               
              <div
                class="flex flex-col gap-2 s-2CpGgZ1sTTPV"
              >
                <div
                  class="s-2CpGgZ1sTTPV"
                >
                  Message
                  <span
                    class="text-red-500 s-2CpGgZ1sTTPV"
                  >
                    *
                  </span>
                </div>
                 
                <textarea
                  class="border-2 s-2CpGgZ1sTTPV"
                  required=""
                  rows="4"
                />
              </div>
               
              <div
                class="flex justify-end s-2CpGgZ1sTTPV"
              >
                <button
                  class="flex justify-around items-center uppercase rounded-sm font-bold w-36 h-12 border border-white shadow-sm mb-2 bg-slate-200  border border-gray-300 text-gray-500"
                  data-testid="contact-business-button"
                  disabled=""
                  id="contact-business-button"
                >
                  Envoyer
                </button>
                <!--&lt;Button&gt;-->
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <!--&lt;+page&gt;-->
  </div>
</body>
`;
